train.py

```diff
@@ -0,0 +1,223 @@
+        D_x = disc_real_output.mean().item()  # Mean discriminator output for real data
+
+        # train generator
+        noise = torch.randn(batch_size, cfg.generator.noise_size, cfg.data.channels, device=cfg.system.device)
+        fake_data = generator(noise)
+        label = fake_labels
+        disc_fake_output = discriminator(fake_data.detach()).view(-1)
+
+        discriminator_error_fake = criterion(disc_fake_output, label)
+        discriminator_error_fake.backward()
+        D_G_z1 = disc_fake_output.mean().item()  # Discriminator's average output when evaluating the fake data
+        discriminator_error = discriminator_error_real + discriminator_error_fake
+        disc_optimizer.step()
+        generator.zero_grad()
+
+        label = real_labels
+        disc_output_after_update = discriminator(fake_data).view(-1)
+        generator_error = criterion(disc_output_after_update, label)
+        generator_error.backward()
+        D_G_z2 = disc_output_after_update.mean().item()  # Discriminator's output after updating the generator, fake data
+        gen_optimizer.step()
+
+        # log to wandb
+        if batch_idx % cfg.train.log_interval == 0:
+            generator_gradients = average_gradient(generator)
+            wandb.log(
+                {
+                    "generator_error": generator_error.item(),
+                    "discriminator_error": discriminator_error.item(),
+                    "D_x": D_x,
+                    "D_G_z1": D_G_z1,
+                    "D_G_z2": D_G_z2,
+                    "generator/generator_gradients": generator_gradients,
+                },
+                commit=False,
+            )
+            fig = visualize_training(generator, fixed_noise, epoch, batch_idx, cfg.logger.chart_path)
+            wandb.log({"fixed noise": wandb.Image(fig)}, commit=True)
+            # TODO: Might add local saving here as well
+            plt.close(fig)
+
+    checkpoint = {
+        "epoch": epoch,
+        "generator_state_dict": generator.state_dict(),
+        "discriminator_state_dict": discriminator.state_dict(),
+        "gen_optimizer_state_dict": gen_optimizer.state_dict(),
+        "disc_optimizer_state_dict": disc_optimizer.state_dict(),
+        "config": cfg,
+        "fixed_noise": fixed_noise,
+    }
+    torch.save(checkpoint, f"{cfg.logger.checkpoint_path}{cfg.run_name}_{epoch}.pt")
+
+
+@hydra.main(version_base=None, config_path="configs", config_name="config")
+def main(cfg: DictConfig):
+    run = wandb.init(
+        project=cfg.project,
+        name=cfg.run_name,
+        job_type="train",
+        config=OmegaConf.to_container(cfg, resolve=True),
+    )
+    set_seed(cfg.system.seed)
+    create_folders(cfg)
+
+    # Initialize models
+    discriminator_net = Discriminator(
+        input_channels=cfg.data.channels,
+        input_size=cfg.data.size,
+        neurons=cfg.discriminator.neurons,
+    ).to(cfg.system.device)
+
+    generator_net = Generator(
```

Comment:
```suggestion
    generator = Generator(
```

We don't need a reminder that it's a net, and this way it will be consistent across the project. Same for the discriminator :)